### PR TITLE
feat(connector): [Paypal] add support for payment and refund webhooks

### PR DESCRIPTION
**Pull request by srujanchikke**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

1) add support for payment and refund webhooks for paypal. As of now source verification is set to false when additional apicall flow for verification is implemented , source verification will be taken care .
2) this pr also modifies connector transaction id for paypal . Before `authorize_id` or `capture_id` are stored in connector transaction id , now it's been changed to `order_id`(Reason : `order_id` is the only constant id that is being sent in webhooks)

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

**Payment Webhook**
Screenshot 2023-08-24 at 1 09 58 PM
**Refund Webhook**
Screenshot 2023-08-25 at 3 03 37 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by ArjunKarthik — APPROVED** [review]

**Comment by ArjunKarthik** [comment]
@prasunna09 Please add the query to move order_id to connector_transaction_id and connector_transaction_id to capture_id in connector_meta_data

Rest looks fine

**Review by ArjunKarthik — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by prasunna09 — APPROVED** [review]

File: crates/router/src/connector/paypal/transformers.rs [hunk 1]
```diff
@@ -853,3 +865,101 @@ pub struct PaypalAccessTokenErrorResponse {
...
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalWebooksEventType {
+    pub event_type: PaypalWebhookEventType,
+}
+
+impl From<PaypalWebhookEventType> for api::IncomingWebhookEvent {
+    fn from(event: PaypalWebhookEventType) -> Self {
+        match event {
+            PaypalWebhookEventType::PaymentCaptureCompleted
+            | PaypalWebhookEventType::PaymentAuthorizationCreated
```
Comment (sai-harsha-vardhan) [inline crates/router/src/connector/paypal/transformers.rs:954]: Why are we marking `PaymentAuthorizationCreated` and `PaymentAuthorizationVoided` as PaymentIntentSuccess?

File: crates/router/src/connector/paypal/transformers.rs [hunk 2]
```diff
@@ -853,3 +865,101 @@ pub struct PaypalAccessTokenErrorResponse {
...
+            | PaypalWebhookEventType::PaymentAuthorizationVoided
+            | PaypalWebhookEventType::CheckoutOrderCompleted => Self::PaymentIntentSuccess,
+            PaypalWebhookEventType::PaymentCapturePending
+            | PaypalWebhookEventType::CheckoutOrderApproved
```
Comment (sai-harsha-vardhan) [inline crates/router/src/connector/paypal/transformers.rs:958]: What's the payment status when we get `CheckoutOrderApproved` or `CheckoutOrderProcessed`

File: crates/router/src/connector/paypal/transformers.rs [hunk 3]
```diff
@@ -203,7 +203,12 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for PaypalPaymentsRequest {
             }
             api::PaymentMethodData::Wallet(ref wallet_data) => match wallet_data {
                 api_models::payments::WalletData::PaypalRedirect(_) => {
-                    let intent = PaypalPaymentIntent::Capture;
+                    let intent = match item.request.is_auto_capture()? {
```
Comment (sai-harsha-vardhan) [inline crates/router/src/connector/paypal/transformers.rs:206]: if statement can be used here

**Review by sai-harsha-vardhan — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/paypal/transformers.rs (hunk 1 above)
Comment (srujanchikke) [reply crates/router/src/connector/paypal/transformers.rs]: This events are moved to event not supported

File: crates/router/src/connector/paypal/transformers.rs (hunk 2 above)
Comment (srujanchikke) [reply crates/router/src/connector/paypal/transformers.rs:959]: reference for status mapping [https://developer.paypal.com/api/rest/webhooks/event-names/](url)

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Comment by Gabby05rose** [comment]
> @prasunna09 Please add the query to move order_id to connector_transaction_id and connector_transaction_id to capture_id in connector_meta_data
> 
> 
> 
> Rest looks fine

Gabby05rose@gmail.com assistance regarding this thank you

**Pull request merged**